core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddressCache.java

```diff
@@ -0,0 +1,42 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import org.infinispan.commons.equivalence.AnyEquivalence;
+import org.infinispan.commons.util.concurrent.jdk8backported.EquivalentConcurrentHashMapV8;
+import org.jgroups.Address;
+import org.jgroups.util.ExtendedUUID;
+import org.jgroups.util.UUID;
+
+/**
+ * Cache JGroupsAddress instances
+ *
+ * @author Dan Berindei
+ * @since 7.0
+ */
+public class JGroupsAddressCache {
+   private static final EquivalentConcurrentHashMapV8<Address, JGroupsAddress> addressCache =
+         new EquivalentConcurrentHashMapV8<>(AnyEquivalence.getInstance(), AnyEquivalence.getInstance());
+
+   static org.infinispan.remoting.transport.Address fromJGroupsAddress(final org.jgroups.Address addr) {
```

Comment:
1) I think I did it because I thought JGroupsTopologyAwareAddress didn't extend JGroupsAddress, but anyway I like returning an interface better.
2) I didn't check the message size, but the memory usage with 40 nodes x 80 caches improved quite a bit, and so did the performance.
3) I only expect the `instanceof` check to be performed about for each other node, so it shouldn't matter.